pkg/kv/kvprober/kvprober.go

```diff
@@ -425,7 +425,7 @@ func (p *Prober) readProbeImpl(ctx context.Context, ops proberOpsI, txns proberT
 
 	d := timeutil.Since(start)
```

Comment:
Does this bug panic the server? We should ensure the call to `logcrash.RecoverAndReportNonfatalPanic` means this bug does NOT panic the server. I think it does, but we should check this via a manual test or similar.

Reply:
the error message does suggest that it panics
`signal_unix.go:881: runtime error: invalid memory address or nil pointer dereference
(1) attached stack trace
  -- stack trace:
  | runtime.sigpanic
  | 	GOROOT/src/runtime/signal_unix.go:881`
will see whats up here

Reply:
Verified locally by panicking the server, but it recovered as expected.